### PR TITLE
Bumped node version and improved tooling

### LINKED ISSUE #1435: [Housekeeping] Use explicit NodeJS version
**Short description of housekeeping request**

As it is right now, we have a set node version for the CI/CD setup (10.16.0), but locally we use different versions, and it's a bit trial and error finding a version that satisfies all dependencies.

I suggest adding a .toolversions file to the repo root. This way developers can use a ASDF https://nodecli.com/nodejs-asdf to automatically manage node version. OR in case a developer opposes this tool, at least it is explicit which Node version we are currently developing up against. It should off course be aligned with CI/CD.

**Additional context**

StencilJS (which we will use for generating web components) has a minimum requirement for NodeJS 12.10.0, but recommends > 14

### Kick Off

* [ ] Ensure this issue has been `Tech refined` with @kirbydesign/kirby-guild and is updated with a clear implementation description
  _This issue should be in the [Ready to do](https://github.com/kirbydesign/designsystem/projects/1#column-4590936) column of the [Kirby kan-ban board](https://github.com/kirbydesign/designsystem/projects/1) before starting implementation_
* [ ] Assign yourself to this issue and move it to the [In progress](https://github.com/kirbydesign/designsystem/projects/1#column-4590937) column of the [Kirby kan-ban board](https://github.com/kirbydesign/designsystem/projects/1)

### Code

* [ ] Create Feature Branch from [master branch](https://github.com/kirbydesign/designsystem/tree/master)
* [ ] Implement/update unit tests
* [ ] Create a draft implementation and push to Github
* [ ] Ask a member of @kirbydesign/kirby-guild for a WIP review by creating a draft Pull Request

### Code Review

* [ ] Open a pull request (or mark the existing draft PR as `Ready for review`) and ask @kirbydesign/kirby-guild for a review
  _Remember to add `closes #issueno` to the description of the PR._
* [ ] Once approved, merge feature branch/PR to master

:tada: Celebrate

### LINKED ISSUE #1443: [Housekeeping] Support for Node LTS (Node 14)
**Short description of housekeeping request**

Support for Node LTS (14.16.0 at this moment).

At the moment Node LTS gives this error:
image

### Kick Off

* [ ] Ensure this issue has been `Tech refined` with @kirbydesign/kirby-guild and is updated with a clear implementation description
  _This issue should be in the [Ready to do](https://github.com/kirbydesign/designsystem/projects/1#column-4590936) column of the [Kirby kan-ban board](https://github.com/kirbydesign/designsystem/projects/1) before starting implementation_
* [ ] Assign yourself to this issue and move it to the [In progress](https://github.com/kirbydesign/designsystem/projects/1#column-4590937) column of the [Kirby kan-ban board](https://github.com/kirbydesign/designsystem/projects/1)

### Code

* [ ] Create Feature Branch from [master branch](https://github.com/kirbydesign/designsystem/tree/master)
* [ ] Implement/update unit tests
* [ ] Create a draft implementation and push to Github
* [ ] Ask a member of @kirbydesign/kirby-guild for a WIP review by creating a draft Pull Request

### Code Review

* [ ] Open a pull request (or mark the existing draft PR as `Ready for review`) and ask @kirbydesign/kirby-guild for a review
  _Remember to add `closes #issueno` to the description of the PR._
* [ ] Once approved, merge feature branch/PR to master

:tada: Celebrate

### PR DESCRIPTION
**Pull request by jakobe**
This PR closes #1443
also closes #1435

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Enhancement (to existing content)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [X] Build related changes
- [X] CI related changes
- [ ] Documentation content changes
- [ ] cookbook application / infrastructure changes
- [ ] Other... Please describe:

## Which issue documents the current behaviour?
Issue Number: #1443, #1435

## What is the new behavior?
* Enforce node version when doing `npm install` through `npmrc` and `engines.node` in `package.json`
* Get node version for CI from `package.json`
* Add support for `nvm install|use` through `.nvmrc`
* Autogenerate `.nvmrc` from `engines.node` in `package.json` when changing node version

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -36,22 +36,23 @@ jobs:
...
-          node-version: ${{ env.NODE_VERSION }}
+          node-version: ${{steps.get_node_version.outputs.NODE_VERSION}}
       - name: Fetch Node Modules Cache
         uses: actions/cache@v2
         id: cache
         with:
           path: | 
             **/node_modules
-          key: node-modules-${{ runner.os }}-${{env.NODE_VERSION}}-${{hashFiles('**/package.json')}}-${{hashFiles('**/package-lock.json')}}
+          key: node-modules-${{ runner.os }}-${{env.NODE_VERSION}}-${{hashFiles('**/package.json', '**/package-lock.json')}}
```
Comment (RasmusKjeldgaard) [inline .github/workflows/build.yml:55]: 🥇  Great boyscouting

**Review by hafstad — APPROVED** [review]
LGTM, however I think it's worth a thought if we should have "^" for our versions of node instead of having an exact version?

**Comment by RasmusKjeldgaard** [comment]
@hafstad Isn't that exactly what is done here? https://github.com/kirbydesign/designsystem/pull/1602/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R6

**Review by RasmusKjeldgaard — APPROVED** [review]
Awesome changes. Took the liberty to change the title ever so slightly, so it looks better on the change log.

**Review by flemcito — APPROVED** [review]
giphub

**Pull request merged**